modules/EvManager/main/car_simulatorImpl.cpp

```diff
@@ -0,0 +1,586 @@
+        simulator->simData->state = SimState::DIODE_FAIL;
+        return true;
+    };
+    commandRegistry->registerCommand("diode_fail", 0, diode_failFunction);
+
+    const auto rcd_currentFunction = [](car_simulatorImpl* const simulator, const std::vector<std::string>& arguments) {
+        simulator->simData->rcd_current_mA = std::stof(arguments[0]);
+        return true;
+    };
+    commandRegistry->registerCommand("rcd_current", 1, rcd_currentFunction);
+
+    const auto iso_wait_slac_matchedFunction = [](car_simulatorImpl* const simulator,
+                                                  const std::vector<std::string>& arguments) {
+        simulator->simData->state = SimState::PLUGGED_IN;
+
+        if (simulator->mod->r_slac.empty()) {
+            EVLOG_debug << "Slac undefined";
+        }
+        if (simulator->simData->slacState == "UNMATCHED") {
+            EVLOG_debug << "Slac UNMATCHED";
+            if (!simulator->mod->r_slac.empty()) {
+                EVLOG_debug << "Slac trigger matching";
+                simulator->mod->r_slac[0]->call_reset();
+                simulator->mod->r_slac[0]->call_trigger_matching();
+                simulator->simData->slacState = "TRIGGERED";
+            }
+        }
+        if (simulator->simData->slacState == "MATCHED") {
+            EVLOG_debug << "Slac Matched";
+            return true;
+        }
+        return false;
+    };
+    commandRegistry->registerCommand("iso_wait_slac_matched", 0, iso_wait_slac_matchedFunction);
+
+    if (!mod->r_ev.empty()) {
+        const auto iso_start_v2g_sessionFunction = [](car_simulatorImpl* const simulator,
+                                                      const std::vector<std::string>& arguments) {
+            const auto& argument = arguments[0];
+            if (argument == "ac_single_phase_core") {
+                simulator->simData->energymode = "AC_single_phase_core";
+            } else if (argument == "ac_three_phase_core") {
+                simulator->simData->energymode = "AC_three_phase_core";
+            } else if (argument == "dc_core") {
+                simulator->simData->energymode = "DC_core";
+            } else if (argument == "dc_extended") {
+                simulator->simData->energymode = "DC_extended";
+            } else if (argument == "dc_combo_core") {
+                simulator->simData->energymode = "DC_combo_core";
+            } else if (argument == "dc_unique") {
+                simulator->simData->energymode = "DC_unique";
+            } else {
+                return false;
+            }
+
+            simulator->mod->r_ev[0]->call_start_charging(simulator->simData->energymode);
+            return true;
+        };
+        commandRegistry->registerCommand("iso_start_v2g_session", 1, iso_start_v2g_sessionFunction);
+    }
+
+    const auto iso_wait_pwr_readyFunction = [](car_simulatorImpl* const simulator,
```

Comment:
Should be also under `if (mod->r_ev.empty())`

Reply:
Done